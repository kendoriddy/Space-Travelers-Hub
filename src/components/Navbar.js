import React from 'react';
import { NavLink } from 'react-router-dom';

const Navbar = () => (
  <nav>
    <ul className="navlist">
      <li>
        <NavLink to="/" className={(navlist) => (navlist.isActive ? 'active' : 'nav-items')} end>
          Rockets
        </NavLink>
      </li>
      <li>
        <NavLink to="/missions" className={(navlist) => (navlist.isActive ? 'active' : 'nav-items')}>
          Missions
        </NavLink>
      </li>
      |
      <li>
        <NavLink to="/profile" className={(navlist) => (navlist.isActive ? 'active' : 'nav-items')}>
          My Profile
        </NavLink>
      </li>
    </ul>
  </nav>
);

export default Navbar;
