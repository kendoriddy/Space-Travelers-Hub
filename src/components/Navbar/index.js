import React from 'react';
import { NavLink } from 'react-router-dom';
import {
  Bars, Nav, NavBtn, NavBtnLink, NavMenu,
} from './NavbarElement';

const Navbar = () => (
  <>
    <Nav>
      <Bars />

      <NavMenu>
        <NavLink to="/about" activeStyle>
          About
        </NavLink>
        <NavLink to="/events" activeStyle>
          Events
        </NavLink>
        <NavLink to="/annual" activeStyle>
          Annual Report
        </NavLink>
        <NavLink to="/team" activeStyle>
          Teams
        </NavLink>
        <NavLink to="/blogs" activeStyle>
          Blogs
        </NavLink>
        <NavLink to="/sign-up" activeStyle>
          Sign Up
        </NavLink>
        {/* Second Nav */}
        {/* <NavBtnLink to='/sign-in'>Sign In</NavBtnLink> */}
      </NavMenu>
      <NavBtn>
        <NavBtnLink to="/signin">Sign In</NavBtnLink>
      </NavBtn>
    </Nav>
  </>
);

export default Navbar;
