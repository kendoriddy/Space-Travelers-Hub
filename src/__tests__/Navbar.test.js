import React from 'react';
import Renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import Navbar from '../components/Navbar';
import store from '../redux/configureStore';

it('renders correctly', () => {
  const tree = Renderer.create(
    <React.StrictMode>
      <Provider store={store}>
        <BrowserRouter>
          <Navbar />
        </BrowserRouter>
      </Provider>
    </React.StrictMode>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
