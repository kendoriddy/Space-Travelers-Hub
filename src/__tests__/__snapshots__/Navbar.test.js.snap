// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as epected 1`] = `null`;

exports[`renders correctly 1`] = `
<nav>
  <ul
    className="navlist"
  >
    <li>
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
      >
        Rockets
      </a>
    </li>
    <li>
      <a
        className="nav-items"
        href="/missions"
        onClick={[Function]}
      >
        Missions
      </a>
    </li>
    |
    <li>
      <a
        className="nav-items"
        href="/profile"
        onClick={[Function]}
      >
        My Profile
      </a>
    </li>
  </ul>
</nav>
`;
